test/helpers/middleware.ts

```diff
@@ -1,12 +1,15 @@
-import { MiddlewareAPI, Dispatch, AnyAction } from '../..'
+import { ThunkDispatch } from 'redux-thunk'
 
-type ThunkAction<T extends any = any> = T extends AnyAction
-  ? AnyAction
-  : T extends Function
-  ? T
-  : never
+import { rootActionTypes } from './actionTypes'
+import { Middleware } from '../..';
 
-export function thunk({ dispatch, getState }: MiddlewareAPI) {
-  return (next: Dispatch) => <T>(action: ThunkAction) =>
+export const thunk: Middleware<
```

Comment:
Unfortunately, using `ThunkMiddleware` kept returning a type of `any` for some reason, so I resorted to this redux `Middleware` type instead, which gives us a properly-typed `thunk`. Worth looking into in the `redux-thunk` library why this is happening, but outside the scope of this PR.